doc: Adds `mongodbatlas_api_key_project_assignment` documentation, examples and migration guide (#3465)
mongodb/terraform-provider-mongodbatlas: docs/data-sources/project_api_key.md

`mongodbatlas_project_api_key` describes a MongoDB Atlas Project API Key. This represents a Project API Key that has been created.

> ~> **IMPORTANT NOTE** The use of `mongodbatlas_project_api_key` data source is no longer the preferred approach. For new configurations, we recommend using the `mongodbatlas_api_key` resource and the `mongodbatlas_api_key_project_assignment` resource to assign the API Keys to projects. This approach is more flexible and aligns with best practices. See the [Migration Guide](../guides/project-api-key-migration.md) for step-by-step instructions on migrating from `mongodbatlas_project_api_key` to the new pattern.

Reviewer: ```suggestion
~> **IMPORTANT NOTE** The use of `mongodbatlas_project_api_key` data source is no longer the recommended approach. For new and existing configurations, we recommend using the `mongodbatlas_api_key` resource and the `mongodbatlas_api_key_project_assignment` resource to assign the API Keys to projects. This approach is more flexible and aligns with best practices. See the [Migration Guide](../guides/project-api-key-migration.md) for step-by-step instructions on migrating from `mongodbatlas_project_api_key` to the new pattern.
```

Author: done in [2771500](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3465/commits/2771500d954e987ce2a03c21417583b162e252a3)

> ~> **IMPORTANT NOTE** The use of `mongodbatlas_project_api_key` data source is no longer the recommended approach. For new and existing configurations, we recommend using the `mongodbatlas_api_key` resource and the `mongodbatlas_api_key_project_assignment` resource to assign the API Keys to projects. This approach is more flexible and aligns with best practices. See the [Migration Guide](../guides/project-api-key-migration.md) for step-by-step instructions on migrating from `mongodbatlas_project_api_key` to the new pattern.

Reviewer: ```suggestion
~> **IMPORTANT NOTE** The use of `mongodbatlas_project_api_key` data source is no longer the recommended approach. For new configurations, we recommend using the `mongodbatlas_api_key` resource and the `mongodbatlas_api_key_project_assignment` resource to assign the API Keys to projects. This approach is more flexible and aligns with best practices. For existing configurations, the migration to the new pattern is **not required**. If you want to migrate, see the [Migration Guide](../guides/project-api-key-migration.md) for step-by-step instructions on migrating from `mongodbatlas_project_api_key` to the new pattern.
```

> ~> **IMPORTANT NOTE** The use of `mongodbatlas_project_api_key` data source is no longer the recommended approach. For new  and existing configurations, we recommend using the `mongodbatlas_api_key` resource and the `mongodbatlas_api_key_project_assignment` resource to assign the API Keys to projects. This approach is more flexible and aligns with best practices. See the [Migration Guide](../guides/project-api-key-migration.md) for step-by-step instructions on migrating from `mongodbatlas_project_api_key` to the new pattern.

Reviewer 2: ```suggestion
~> **IMPORTANT NOTE** The use of `mongodbatlas_project_api_key` data source is no longer the recommended approach. For new and existing configurations, we recommend using the `mongodbatlas_api_key` resource and the `mongodbatlas_api_key_project_assignment` resource to assign the API Keys to projects. This approach is more flexible and aligns with best practices. See the [Migration Guide](../guides/project-api-key-migration.md) for step-by-step instructions on migrating from `mongodbatlas_project_api_key` to the new pattern.
```

Author: fixed in [24fc1f8](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3465/commits/24fc1f85893b7ce14d2f0b26d12ce57ae66f056d). Thanks
